### PR TITLE
Provide a way to modify the text to embed

### PR DESCRIPTION
**Pull request by pparmar30**
Resolves https://github.com/aryn-ai/sycamore/issues/130

### CODE REVIEW

**Review by HenryL27 — COMMENTED** [review]
LGTM, Thanks!
I noticed that bedrock (unbatched) and ST embedders handle null `text_representation`s differently. Can we follow the bedrock logic for sentence transformers? (if no text representation, just skip) Not sure how that would integrate w the preprocess fn.

File: sycamore/transforms/embed.py [hunk 1]
```diff
@@ -19,17 +19,23 @@
 from sycamore.utils import batched, generate_map_batch_function, generate_map_batch_class_from_callable
 
 
+def _pre_process_document(document: Document) -> str:
+    return document.text_representation if document.text_representation is not None else ""
```
Comment (bsowell) [inline sycamore/transforms/embed.py:23]: I would agree with what Henry said. Some of the embedding models will break on empty strings (which is why I did it differently for OpenAI/Bedrock). Maybe better to skip empty passages.

Comment (bsowell) [reply sycamore/transforms/embed.py:23]: Maybe could do something as simple as if you return null from the preprocess function then skip embedding the element.

File: sycamore/transforms/embed.py [hunk 2]
```diff
@@ -90,7 +96,7 @@ def generate_embeddings(self, doc_batch: list[Document]) -> list[Document]:
 
         assert self._transformer is not None
 
-        text_batch = ["" if doc.text_representation is None else doc.text_representation for doc in doc_batch]
+        text_batch = [self.pre_process_document(doc) for doc in doc_batch if doc.text_representation is not None]
         embeddings = self._transformer.encode(text_batch, batch_size=self.model_batch_size, device=self.device)
         for doc, embedding in zip(doc_batch, embeddings):
```
Comment (HenryL27) [inline sycamore/transforms/embed.py:101]: well now doc_batch and embeddings could be different lengths bc maybe a doc got take out of the middle, so won't that cause potential errors?

Comment (pparmar30) [reply sycamore/transforms/embed.py]: good catch! Updated

**Review by HenryL27 — APPROVED** [review]

**Pull request merged**